syntax = "proto2";
package DB.Protos;

option cc_generic_services = true;

message TransferRuntimeFilterRequest {
  required string query_id = 1;
  required uint64 filter_id = 2;
  // optional uint64 execute_segment_id = 3;
  required string remote_address = 4;
  required uint64 require_parallel_size = 5;
  required bytes filter_data = 6;

  //      optional uint64 build_segment_id = 2;
}

message TransferRuntimeFilterResponse {
  optional string message = 1;
}

message DispatchRuntimeFilterRequest {
  required string query_id = 1;
  required uint64 segment_id = 2;
  required uint64 filter_id = 3;
  required bytes filter_data = 4;
}

message DispatchRuntimeFilterResponse {
  optional string message = 1;
}

service RuntimeFilterService {

  rpc transferRuntimeFilter(TransferRuntimeFilterRequest) returns (TransferRuntimeFilterResponse);

  rpc dispatchRuntimeFilter(DispatchRuntimeFilterRequest) returns (DispatchRuntimeFilterResponse);
};

